### PR TITLE
[kudu] Update Kudu binding to 1.1.0, use slf4j for logging, improve table partitioning

### PR DESCRIPTION
**Pull request by danburkert**
This updates Kudu to the latest released client version, improves table partitioning, and switches debug logging to use slf4j.  Further details are in the commit messages.

### CODE REVIEW

File: kudu/pom.xml [hunk 1]
```diff
@@ -41,18 +41,23 @@ LICENSE file.
...
-      <releases>
-        <enabled>true</enabled>
-      </releases>
-      <snapshots>
-        <enabled>false</enabled>
-      </snapshots>
-      <id>cloudera-repo</id>
-      <name>Cloudera Releases</name>
-      <url>https://repository.cloudera.com/artifactory/cloudera-repos</url>
+      <id>apache.releases</id>
```
Comment (risdenk) [inline kudu/pom.xml]: Does this need to be there? Can the kudu-client be pulled from Maven central?

File: kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java [hunk 2]
```diff
@@ -34,67 +38,51 @@
...
-      this.printErrors =
-          getProperties().getProperty(PRINT_ROW_ERRORS_OPT).equals("true");
-    }
-    if (getProperties().getProperty(PRINT_ROW_ERRORS_OPT) != null) {
-      this.printErrors =
-          getProperties().getProperty(PRINT_ROW_ERRORS_OPT).equals("true");
-    }
-    this.tableName = com.yahoo.ycsb.workloads.CoreWorkload.table;
-    initClient(debug, tableName, getProperties());
+    String tableName = CoreWorkload.table;
```
Comment (risdenk) [inline kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java]: This assumes the CoreWorkload is being used?

File: kudu/src/main/java/com/yahoo/ycsb/db/package-info.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
  */
 
 /**
- * The YCSB binding for <a href="http://getkudu.io/">Kudu</a>.
+ * The YCSB binding for <a href="http://kudu.apache.org/">Apache Kudu</a>.
```
Comment (risdenk) [inline kudu/src/main/java/com/yahoo/ycsb/db/package-info.java]: Copyright should be updated.

File: kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java [hunk 4]
```diff
@@ -23,9 +23,13 @@
 import com.yahoo.ycsb.Status;
 import com.yahoo.ycsb.StringByteIterator;
 import com.yahoo.ycsb.workloads.CoreWorkload;
-import org.kududb.ColumnSchema;
-import org.kududb.Schema;
-import org.kududb.client.*;
+import org.slf4j.Logger;
```
Comment (risdenk) [inline kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java]: The header on this file doesn't have the YCSB copyright?

File: kudu/pom.xml (hunk 1 above)
Comment (danburkert) [inline kudu/pom.xml]: No, it's currently only published to the Apache maven repo.

File: kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java (hunk 2 above)
Comment (danburkert) [inline kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java]: Yes, I think it does.  This behavior is not changing as part of this PR.

File: kudu/pom.xml (hunk 1 above)
Comment (danburkert) [inline kudu/pom.xml]: Awesome, I was not aware of that.

File: kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java (hunk 4 above)
Comment (danburkert) [inline kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java]: I don't know the copyright details on this file.

File: kudu/src/main/java/com/yahoo/ycsb/db/package-info.java (hunk 3 above)
Comment (danburkert) [inline kudu/src/main/java/com/yahoo/ycsb/db/package-info.java]: done.

**Comment by danburkert** [comment]
> Will these changes prevent someone comparing Kudu <1.x with this updated client?

For the most part.  These changes may be compatible with 0.10 (the last release before 1.0), but I haven't checked.

> Is there any reason someone would want to compare <1.x with >1.x?

Probably not.  Now that Kudu is 1.0 there should be no such breaking changes in the future.

**Comment by danburkert** [comment]
rebased and squashed fixup commits.

File: kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java (hunk 4 above)
Comment (risdenk) [inline kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java]: Should be above the license with a line like:

```
Copyright (c) 2016 YCSB contributors. All rights reserved.
```

See `kudu/README.md` for an example.

File: kudu/src/main/java/com/yahoo/ycsb/db/package-info.java (hunk 3 above)
Comment (risdenk) [inline kudu/src/main/java/com/yahoo/ycsb/db/package-info.java]: Whoops sorry meant to throw an example in here:

```
Copyright (c) 2015-2016 YCSB contributors. All rights reserved.
```

File: kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java (hunk 2 above)
Comment (risdenk) [inline kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java]: O yea didn't see that it hadn't changed.

File: kudu/README.md [hunk 5]
```diff
@@ -17,25 +17,30 @@ LICENSE file.
 
 # Kudu bindings for YCSB
 
-[Kudu](http://getkudu.io) is a storage engine that enables fast analytics on fast data.
+[Apache Kudu](https://kudu.apache.org) is a storage engine that enables fast
```
Comment (risdenk) [inline kudu/README.md]: Looks like the header on this file should be updated as well:

```
Copyright (c) 2015-2016 YCSB contributors. All rights reserved.
```

**Comment by risdenk** [comment]
@danburkert Looks pretty good to me. Just minor header changes and then can merge it.

**Comment by danburkert** [comment]
Added a commit fixing up the copyrights.

**Pull request merged**